lib/monitor/measure.go

```diff
@@ -36,124 +75,143 @@ func ConnectSink(addr string) error {
 		return err
 	}
 	dbg.Lvl3("Connected to sink:", addr)
+	sink = addr
 	connection = conn
 	encoder = json.NewEncoder(conn)
 	return nil
 }
 
-// Only sends a ready-string
-func Ready(addr string) error {
-	if encoder == nil {
-		dbg.Lvl3("Connecting to sink", addr)
-		err := ConnectSink(addr)
-		if err != nil {
-			return err
-		}
+// NewSingleMeasure returns a new measure freshly generated
+func NewSingleMeasure(name string, value float64) *SingleMeasure {
+	return &SingleMeasure{
+		Name:  name,
+		Value: value,
 	}
-	dbg.Lvl3("Sending ready-signal")
-	send(Measure{Name: "ready"})
-	return nil
 }
 
-// Returns how many peers are ready
-func GetReady(addr string) (*Stats, error) {
-	if encoder == nil {
-		err := ConnectSink(addr)
-		if err != nil {
-			return nil, err
-		}
+// Record sends the value to the monitor. Reset the value to 0.
+func (sm *SingleMeasure) Record() {
+	if err := send(sm); err != nil {
+		dbg.Error("Error sending SingleMeasure", sm.Name, " to monitor:", err)
 	}
-	dbg.Lvl3("Getting ready_count")
-	send(Measure{Name: "ready_count"})
-	decoder := json.NewDecoder(connection)
-	var s Stats
-	err := decoder.Decode(&s)
-	if err != nil {
-		return nil, err
+	sm.Value = 0
+}
+
+// NewTimeMeasure return *TimeMeasure
+func NewTimeMeasure(name string) *TimeMeasure {
+	tm := &TimeMeasure{name: name}
+	tm.reset()
+	return tm
+}
+
+// Send measurements to the monitor:
+// - wall time: *name*_wall
+// - system time: *name*_system
+// - user time: *name*_user
+func (tm *TimeMeasure) Record() {
+	// Wall time measurement
+	tm.Wall = NewSingleMeasure(tm.name+"_wall", float64(time.Since(tm.lastWallTime))/1.0e9)
+	// CPU time measurement
+	tm.Cpu.Value, tm.User.Value = getDiffRTime(tm.Cpu.Value, tm.User.Value)
+	// send data
+	tm.Wall.Record()
+	tm.Cpu.Record()
+	tm.User.Record()
+	// reset timers
+	tm.reset()
+
+}
+
+// reset reset the time fields of this time measure
+func (tm *TimeMeasure) reset() {
+	cpuTimeSys, cpuTimeUser := getRTime()
+	tm.Cpu = NewSingleMeasure(tm.name+"_system", cpuTimeSys)
+	tm.User = NewSingleMeasure(tm.name+"_user", cpuTimeUser)
+	tm.lastWallTime = time.Now()
+}
+
+// CounterIO is an interface that can be used to count how many bytes does an
+// object have written and how many bytes does it have read. For example it is
+// implemented by cothority/network/ Conn  + Host to know how many bytes a
+// connection / Host has written /read
+type CounterIO interface {
+	Rx() uint64
+	Tx() uint64
```

Comment:
`Tx` and `Rx` need a comment. Also mention what they stand for (transmitted and received x ...?)